include/dxc/Support/WinAdapter.h

```diff
@@ -15,8 +15,70 @@
 #ifndef LLVM_SUPPORT_WIN_ADAPTER_H
 #define LLVM_SUPPORT_WIN_ADAPTER_H
 
-#ifndef _WIN32
+#ifdef _WIN32
+#ifndef CROSS_PLATFORM_UUIDOF
+#ifndef __MINGW32__
+#define CROSS_PLATFORM_UUIDOF(interface, spec)                                 \
+  struct __declspec(uuid(spec)) interface;
+#else // __MINGW32__
+#include <guiddef.h>
+#include <sal.h>
+// The following macros may not present in mingw's `sal.h` header
+#ifndef _In_count_
+#define _In_count_(s)
+#endif
+#ifndef _In_opt_count_
+#define _In_opt_count_(s)
+#endif
+#ifndef _In_bytecount_
+#define _In_bytecount_(s)
+#endif
+#ifndef _Maybenull_
+#define _Maybenull_
+#endif
+constexpr unsigned char nybble_from_hex(char c) {
```

Comment:
I don't think you can use constexpr outside of the __cplusplus guard, no?

Reply:
Good catch, you are right

Reply:
__CRT_UUID_DECL only for C++, and define to nothing for C